### DISCUSSION TITLE
对端何时知道本端新建了一个stream ？

### DISCUSSION THREAD

**Discussion by stopandjoin** [Tencent/tquic#210] [Q&A (问题和回答)] [answered]
发现在一端 quic_stream_new 后， 只有本端才会有 stream created回调。  对端察觉不到，需要本端往这个stream中写入数据后， 对端才会有stream created回调。  Tquic中是如此设计的吗？

**Comment by glime-ai** [comment, marked as answer]
是的，根据TQUIC的设计和文档描述，流（Stream）的创建是由一端主动发起的，而对端确实需要等待有数据写入该流后才能觉察到新流的创建。这是因为在QUIC协议中，流的创建是隐式的，即当一端向另一端发送关于某个尚未被对方知晓的流ID的数据时，接收端会认为该流已被创建。因此，在TQUIC中，一端使用`quic_stream_new`方法创建新流并不会立即在对端触发`stream created`回调，只有当数据被写入并发送到该流时，对端才会觉察到新流的创建并触发相应的回调。

这种设计允许流的创建过程更加灵活和高效，避免了在数据传输之前需要进行额外的流创建通信，从而减少了延迟和开销。同时，这也意味着应用程序需要在发送数据之前，确保对端已经准备好接收和处理新流的数据。
